src/Mvc/Mvc.Abstractions/src/ModelBinding/ModelMetadata.cs

```diff
@@ -521,6 +536,22 @@ internal void ThrowIfRecordTypeHasValidationOnProperties()
         }
     }
 
+    internal static Func<ParameterExpression, Expression, Expression>? FindTryParseMethod(Type modelType)
+    {
+        try
+        {
+            modelType = Nullable.GetUnderlyingType(modelType) ?? modelType;
+            return ParameterBindingMethodCache.FindTryParseMethod(modelType);
+        }
+        catch (InvalidOperationException)
+        {
+            // The ParameterBindingMethodCache.FindTryParseMethod throws an exception
```

Comment:
Suggestion - we could update `FindTryParseMethod` to not throw via an additional parameter. Not catching exceptions is certainly a lot nicer since it avoids things like first-chance exceptions in VS.

Reply:
Agree. I will open a new PR just for that, ok?